src/index.js

```diff
@@ -48,14 +49,35 @@ class Service {
 
   create (body, params = {}) {
     let { id, uri } = body;
-    const { buffer, MIME: contentType } = parseDataURI(uri);
-    const hash = bufferToHash(buffer);
-    const ext = mimeTypes.extension(contentType);
+    let file = params.file;
+    let hash, ext, stream;
+    let size = null;
+
+    if (uri) {
+      const { buffer, MIME: contentType } = parseDataURI(uri);
+      hash = bufferToHash(buffer);
+      ext = mimeTypes.extension(contentType);
+      stream = fromBuffer(buffer);
+      size = buffer.length;
+    } else if (file && file.buffer) {
+      const buffer = params.file.buffer;
+      const contentType = params.file.mimetype
+      hash = bufferToHash(buffer);
+      ext = mimeTypes.extension(contentType);
+      stream = fromBuffer(buffer);
+      size = file.size;
+    } else if (file && file.path) {
+      const contentType = params.file.mimetype
+      hash = params.file.filename;
+      ext = mimeTypes.extension(contentType);
+      stream = fs.createReadStream(params.file.path);
+      size = file.size;
+    }
```

Comment:
If it doesn't hit any of those branches it should probably throw some kind of error.